content/entry-hash/index.md

```diff
@@ -0,0 +1,219 @@
+
+```json
+[
+  {
+    "index-entry-number":"6",
+    "entry-number":"6",
+    "entry-timestamp":"2016-04-05T13:23:05Z",
+    "key":"GB",
+    "item-hash":["sha-256:6b18693874513ba13da54d61aafa7cad0c8f5573f3431d6f1c04b07ddb27d6bb"]
+  }
+]
+```
+
+Notice they are not exaclty the same even thought they represent the same
+entity. Also notice that the `ID` type is not yet an approved RFC. For this
+RFC this shouldn't be a problem as it works the same as if the type were a
+string.
+
+Each value MUST be tagged by a byte that identifies the type, in a similar way
+as the [objecthash](https://github.com/benlaurie/objecthash) algorithm.
+
+```elm
+type Tag
+  = Hash
+  | Integer
+  | Set
+  | String
+  | Timestamp
+  | List
+
+tagToChar : Tag -> Char
+tagToChar tag =
+  case tag of
+    Hash ->
+      'r'
+    Integer ->
+      'i'
+    Set ->
+      's'
+    String ->
+      'u'
+    Timestamp ->
+      't'
+    List ->
+      'l'
+```
+
+```elm
+hashValue : Alg -> String -> Tag -> Hash
+```
+
+Where:
+
+* `Alg` is the hashing algorithm (e.g. `Sha256`),
+* `String` is the string representation of the value to hash,
+* `Tag` is the tag for the `Value` type and
+* `Hash` is the list of bytes resulting of the hashing algorithm.
+
+Following the JSON example from above, these are the hashes for each value:
+
+```elm
+numberValue = toString 6
+numberValue == "6"
+
+numberHash = hashValue Sha256 numberValue Integer
+toHex numberHash == "396ee89382efc154e95d7875976cce373a797fe93687ca8a27589116644c4bcd"
+```
+
+```elm
+timestampValue = toString (Timestamp 2016 4 5 13 23 5 Utc)
+toHex timestampValue == "2016-04-05T13:23:05Z"
+
+timestampHash = hashValue Sha256 timestampValue Timestamp
+toHex timestampHash == "f22ecc4464f22c8fee624769189665a0afd7ef10a2775a000082c47cbd9f6419"
+```
+
+```elm
+keyValue = toString (ID "GB")
+keyValue == "GB"
+
+keyHash = hashValue Sha256 keyValue String
+toHex keyHash == "fff7021c7df4426be0f9a3c83f236eb6f85d159e624b010d65e6dde267889c21"
+```
+
+The item is a special case:
+
+It is a set of values and each value is already hashed. This means that the
+content of this set only needs a sorted concatenation. Then, tag it as a set
+`s` and hash the result.
+
+```elm
+hashValueSet : Alg -> Set Hash -> Tag -> Hash
+
+itemHash = hashValueSet Sha256 (Set [Sha256 "6b18693874513ba13da54d61aafa7cad0c8f5573f3431d6f1c04b07ddb27d6bb"]) Hash
+toHex itemHash == "94b34bd5705b1a8c31796c00347dad2f5a1364050a1f6b768c7ac289502f62ce"
+```
+
+### order the resulting hashes
+
+The order is:
+
+```elm
+hashList =
+  [ numberHash
+  , keyHash
+  , timestampHash
+  , itemHash
+  ]
+```
+
+Note: The order has been chosen arbitrarily.
+
+### hash the result of concatenating all hashes
+
+And finally, tag and hash the result of concatenating the list of hashes:
+
+```elm
+hash : Alg -> List Hash -> Hash
+
+entryHash = hash Sha256 hashList
+entryHash == "cc33f805025fb5bcd43cd9a1d0d77093af443106ff1513a2f5dfff2b81ae4fd3"
```

Comment:
I also can't get this result from trying the example? Can you clarify, perhaps in code, how you did it?

Reply:
I managed to get this. I was just doing it wrong. So ignore this comment!